release/release.py

```diff
@@ -131,18 +132,28 @@ def main() -> None:
     """Create the release."""
     os.environ["RELEASE_DATE"] = datetime.date.today().isoformat()  # Used by bump-my-version to update CHANGELOG.md
     bump, current_version, check_preconditions_only = parse_arguments()
-    check_preconditions(bump, current_version)
-    if check_preconditions_only:
-       return
     # See https://github.com/callowayproject/bump-my-version?tab=readme-ov-file#add-support-for-pre-release-versions
     # for how bump-my-version deals with pre-release versions
+    create_rc = False
     if bump.startswith("rc-"):
+        create_rc = True
```

Comment:
I think the flag should also be true when bumping a rc. When the current release is a rc, `create_rc` would not be set to `True`. This means the version history will be checked for a release date which is not needed since we're creating another release candidate:

```suggestion
    create_rc = False
    if bump == "rc" or bump.startswith("rc-"):
        create_rc = True
```